math-scala/src/main/scala/org/apache/mahout/math/algorithms/regression/Regressor.scala

```diff
@@ -0,0 +1,33 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one
+  * or more contributor license agreements. See the NOTICE file
+  * distributed with this work for additional information
+  * regarding copyright ownership. The ASF licenses this file
+  * to you under the Apache License, Version 2.0 (the
+  * "License"); you may not use this file except in compliance
+  * with the License. You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing,
+  * software distributed under the License is distributed on an
+  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  * KIND, either express or implied. See the License for the
+  * specific language governing permissions and limitations
+  * under the License.
+  */
+
+package org.apache.mahout.math.algorithms.regression
+
+import org.apache.mahout.math.algorithms.Model
+import org.apache.mahout.math.drm.DrmLike
+
+/**
+  * Abstract of Regressors
+  */
+abstract class Regressor extends Model {
+
+  def fit[Int](drmY: DrmLike[Int], drmX: DrmLike[Int]): Unit
```

Comment:
like perhaps `fit[K](X:drmLike[K],T:drmLike[K], (Symbol,Any)*):Model` where the optional list is hyperparameter list. Then hyperparameterized calls could be something like: ```
fit(X,Y, 'alpha ->10.0, 'lambda -> 1.e-15)
``` etc.
This loses strong type-iness of the signature but call is not that ugly, and it might be ok if specific implementations are cleanly documented .